### DISCUSSION TITLE
Need initial setup guidance

### DISCUSSION THREAD

**Discussion by Arelius-D** [qdm12/ddns-updater#713] [Q&A] [answered]
Hey guys,

Would someone be so kind and help me out here to install and run this as a container?
_I recently learned to how install docker and docker-compose. I know how to create docker-compose.yml and understand the basics of configure such file and the run 'docker compose up -d' (so the container runs in the background)._

I thought this packaged here could be executed in the same manner, no? .._maybe it can and I just cant find the documentation for it and so I fail over and over again, or it follows some other process that I'm unfamiliar with_

I (think) I have manged to pull the image
`docker pull qmcgaw/ddns-updater`

Which I check using 'docker images'
`qmcgaw/ddns-updater       latest       056010abbb0f       22 hours ago       16.4MB`

Would someone like to help me out here and install it?

Your input would be much appreciated!

**Comment by qdm12** [comment]
See https://github.com/qdm12/ddns-updater?tab=readme-ov-file#container

You basically replace step 3 with https://github.com/qdm12/ddns-updater/blob/master/docker-compose.yml (it is mentioned in step 4) 

Are you blocked at any step?

**Reply by Arelius-D** [reply]
I did, actually even tried.. Got confused by step 1 and 2, jumped step 3.. failed ofc.
Found the docker-compose.yml and failed again, ofc..
So I gathered I need to follow step 1 and 2 and when failed in doing so I figured maybe I can try and ask for some help so I wrote here asking for help.

This is what I read, "Step 1, create a folder called _data_ and put following code inside a file called _config.json_"
_and so I did mkdir data -> cd data -> nano config.json and inserted this block of code:_
```
mkdir data
touch data/config.json
# Owned by user ID of Docker container (1000)
chown -R 1000 data
# all access (for creating json database file data/updates.json)
chmod 700 data
# read access only
chmod 400 data/config.json
```
"step 2, inside the same file located in the same folder put this in:"
```
{
    "settings": [
        {
            "provider": "namecheap",
            "domain": "example.com",
            "host": "@",
            "password": "e5322165c1d74692bfa6d807100c0310"
        }
    ]
}
```
So I did just that.. nano config.json .. which is obviously wrong cause I'm opening and editing the file created in step one _I'm not mentally challenged but I really don't understand_

So I'm stuck on step 1 and 2 cause it doesn't make seance that the same file in the same folder first get one input and then be overwritten in the next step by something else.


Ps, I have NoIP as provider so I specified 'config.jason' like this:


`cat data/config.json`

```
{
  "settings": [
    {
      "provider": "noip",
      "domain": "ddns.net",
      "host": "@",
      "username": "not_telling",
      "password": "not_telling",
      "ip_version": "ipv4",
      "ipv6_suffix": "",
      "provider_ip": true
    }
  ]
}
```

**Reply by Arelius-D** [reply]
Ok I figured out that the "_code_" below step 1 are **not to be inserted into the config.jason** file, they are there to set and ensure correct permissions!

So, moving on from step 1 (since its all set and done)

I used the docker-compose.yml, dint touch anything in it beside the restart value to:` unless-stopped`

So step 2, he is current config.json:

```
{
  "settings": [
    {
      "provider": "noip",
      "domain": "ddns.net",
      "host": "NOTTELLING.ddns.net",
      "username": "not_telling",
      "password": "not_telling",
      "ip_version": "ipv4",
      "ipv6_suffix": "",
      "provider_ip": true
    }
  ]
}
```
`docker compose up -d` gives me

```WARN[0000] /home/pi/docker/ddns-updater/docker-compose.yml: `version` is obsolete
[+] Running 1/1
 ✔ Container ddns-updater  Started```  

and here is the log by running;
`docker logs ddns-updater`

```
========================================
========================================
============= ddns-updater =============
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2024-05-06T11:16:06.246Z (commit 542e895)

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2024-05-07T21:18:51Z INFO Settings summary:
├── HTTP client
|   └── Timeout: 10s
├── Update
|   ├── Period: 5m0s
|   └── Cooldown: 5m0s
├── Public IP fetching
|   ├── HTTP enabled: yes
|   ├── HTTP IP providers
|   |   └── all
|   ├── HTTP IPv4 providers
|   |   └── all
|   ├── HTTP IPv6 providers
|   |   └── all
|   ├── DNS enabled: yes
|   ├── DNS timeout: 3s
|   └── DNS over TLS providers
|       └── all
├── Resolver: use Go default resolver
├── Server
|   ├── Listening address: :8000
|   └── Root URL: /
├── Health
|   └── Server listening address: 127.0.0.1:9999
├── Paths
|   └── Data directory: /updater/data
├── Backup: disabled
└── Logger
    ├── Level: info
    └── Caller: hidden
2024-05-07T21:18:51Z INFO reading JSON config from file /updater/data/config.json
2024-05-07T21:18:51Z INFO Found single setting to update record
2024-05-07T21:18:51Z INFO Reading history from database: domain ddns.net host NOTTELLING.ddns.net ipv4
2024-05-07T21:18:51Z INFO [backup] disabled
2024-05-07T21:18:51Z INFO [healthcheck server] listening on 127.0.0.1:9999
2024-05-07T21:18:51Z INFO [http server] listening on :8000
2024-05-07T21:18:51Z WARN cannot DNS resolve NOTTELLING.ddns.net.ddns.net after 5 tries: lookup NOTTELLING.ddns.net.ddns.net on 10.0.0.1:53: no such host
2024-05-07T21:18:51Z INFO ipv4 address of NOTTELLING.ddns.net.ddns.net is invalid IP and your ipv4 address  is CORRECT_IP
2024-05-07T21:18:51Z INFO Updating record [domain: ddns.net | host: NOTTELLING.ddns.net | provider: noip | ip: ipv4] to use CORRECT_IP
2024-05-07T21:18:52Z ERROR unknown response received: nohost
```
Whats have I missed? Cause according to the loggs `WARN cannot DNS resolve NOTTELLING.ddns.net.ddns.net after 5 tries: lookup NOTTELLING.ddns.net.ddns.net on 10.0.0.1:53: no such host` so there is something I have done wrong here it feels..

**Reply by danvandam01** [reply]
i'm struggling to figure out where to create the data directory.  I'm running portainer on a raspberry pi.  I've created a directory but I created it here: /home/myname/updater/data.  I'm getting the log error 
2024-09-06T03:21:35Z INFO reading JSON config from file /updater/data/config.json
2024-09-06T03:21:35Z INFO file not found, creating an empty settings file

So clearly i've created the updater/data file in the wrong place.  Any help would be appreciated.

**Comment by Arelius-D** [comment, marked as answer]
Ok for NoIP configuration in the config.json file
```
{
  "settings": [
    {
      "provider": "noip",
      "domain": "[only the domain name, eg ddns.net]",
      "host": "[only the host name and not the entire adress which is, hostname + domain]",
      "username": "[your username/accountname @ noip.com]",
      "password": "[your account password @ noip.com]",
      "ip_version": "ipv4",
      "ipv6_suffix": "",
      "provider_ip": true
    }
  ]
}
```

`docker logs ddns-updater`

```
========================================
========================================
============= ddns-updater =============
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2024-05-06T11:16:06.246Z (commit 542e895)

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2024-05-07T21:30:48Z INFO Settings summary:
├── HTTP client
|   └── Timeout: 10s
├── Update
|   ├── Period: 5m0s
|   └── Cooldown: 5m0s
├── Public IP fetching
|   ├── HTTP enabled: yes
|   ├── HTTP IP providers
|   |   └── all
|   ├── HTTP IPv4 providers
|   |   └── all
|   ├── HTTP IPv6 providers
|   |   └── all
|   ├── DNS enabled: yes
|   ├── DNS timeout: 3s
|   └── DNS over TLS providers
|       └── all
├── Resolver: use Go default resolver
├── Server
|   ├── Listening address: :8000
|   └── Root URL: /
├── Health
|   └── Server listening address: 127.0.0.1:9999
├── Paths
|   └── Data directory: /updater/data
├── Backup: disabled
└── Logger
    ├── Level: info
    └── Caller: hidden
2024-05-07T21:30:48Z INFO reading JSON config from file /updater/data/config.json
2024-05-07T21:30:48Z INFO Found single setting to update record
2024-05-07T21:30:48Z INFO Reading history from database: domain ddns.net host NOTTELLING ipv4
2024-05-07T21:30:48Z INFO [backup] disabled
2024-05-07T21:30:48Z INFO [http server] listening on :8000
2024-05-07T21:30:48Z INFO [healthcheck server] listening on 127.0.0.1:9999
```

So it seems to be working now!
